android/src/main/java/app/notifee/core/utility/PowerManagerUtils.java

```diff
@@ -93,6 +93,26 @@ public static Boolean isBatteryOptimizationEnabled(Context context) {
     return !pm.isIgnoringBatteryOptimizations(context.getPackageName());
   }
 
+  /**
+   * Lights up the screen if it is not already lit
+   *
+   * @param context
+   */
+  public static void lightUpScreenIfNeeded(Context context) {
+    PowerManager pm = (PowerManager) context.getSystemService(Context.POWER_SERVICE);
+    Boolean isScreenOn = pm.isScreenOn();
+
+    if (isScreenOn == false) {
+        PowerManager.WakeLock wl = pm.newWakeLock(
+          PowerManager.FULL_WAKE_LOCK |PowerManager.ACQUIRE_CAUSES_WAKEUP |PowerManager.ON_AFTER_RELEASE, "MyLock"
```

Comment:
nit: can we change the tag string such as 'Notifee:lock' and 'Notifee:cpuLock'?

Also, curious why do we need both `FULL_WAKE_LOCK ` and `PARTIAL_WAKE_LOCK `?

Reply:
About the name, absolute, I was so focused on the functionality that I forgot about the name rs, I'll change it as you recommended.

About the `FULL_WAKE_LOCK` and `PARTIAL_WAKE_LOCK` are usually used together because:
`FULL_WAKE_LOCK`: Wake lock level: Ensures that the screen and keyboard backlight are on at full brightness.
`PARTIAL_WAKE_LOCK`: Wake lock level: Ensures that the CPU is running; the screen and keyboard backlight will be allowed to go off.

So basically `FULL_WAKE_LOCK` lights up the screen and `PARTIAL_WAKE_LOCK` the CPU.

[Android Docs](https://developer.android.com/reference/android/os/PowerManager#FULL_WAKE_LOCK)


**Observation:** The `FULL_WAKE_LOCK` is deprecated and Android documentation recommends us to use [flags](https://developer.android.com/reference/android/view/WindowManager.LayoutParams#FLAG_KEEP_SCREEN_ON) with the intent to keep it awake. But I confess that with my current time I could not explore `notifee` lib enough to find the best place to apply and test it. So I think is good to know about this deprecation and maybe in the future try to change it to the FLAG approach.

The approach with FLAG looks to make the intent keep awake, in our case maybe it will not work as we want since we have the `lightUpScreen` param to make it happen, anyway, it's something that really need some extra attention in the future, sorry by no providing more information

Reply:
Hmm :thinking: from the PARTIAL_WAKE_LOCK docs:

> If the user presses the power button, then the screen will be turned off but the CPU will be kept on until all partial wake locks have been released.

I have to admit, as a user with notifications on my personal phone, I expect to be able to hit the power button and have the screen go off - I may not be responding to the notification but I've seen it at that point and I want the phone to be "quiet" (visually) again.

Would this PR work (screen go on, at least temporarily) if only PARTIAL_WAKE_LOCK was used? If so, then it avoids the FULL_WAKE_LOCK permission + deprecation and simplifies future work (moving to flags) plus user experience (permissions) in one swoop, possibly with a better user experience (tap power button to quiet phone is possible)

Reply:
Hmm unfortunately this PR is not going to work just with `PARTIAL_WAKE_LOOK`, as the docs say:
> Wake lock level: Ensures that the CPU is running; the screen and keyboard backlight will be allowed to go off.

`PARTIAL_WAKE_LOOK` just ensures the CPU side, just with that the screen does not lights up, I confirmed it testing it here and other JAVA/Android developers said the same in some Stackoverflow and other forums threads :/

And by what I understood from the docs, the use of `FULL_WAKE_LOOK` and `PARTIAL_WAKE_LOOK` together will let the user tap the power button to quiet/erase the screen but the CPU will be kept on until all partial wake locks have been released.

Reply:
Yes, would be nice if it work just with `PARTIAL_WAKE_LOCK`.

Yeah, at least one more year (I believe even more) it's going to work fine, maybe is time enough to prioritize it and get a final solution

Reply:
@gustavogialim thanks for the explanation and yeah agree with @mikehardy it's a bit weird to not be able to hit the power button and switch the screen off... it's good to have something in the lib so developers have the option to use this or not.